apps/shipments/views.py

```diff
@@ -22,12 +23,26 @@ class ShipmentViewSet(viewsets.ModelViewSet):
     serializer_class = ShipmentSerializer
     permission_classes = ((permissions.IsAuthenticated, IsOwner) if settings.PROFILES_ENABLED
                           else (permissions.AllowAny,))
+    filter_backends = (filters.SearchFilter, filters.OrderingFilter, DjangoFilterBackend,)
+    searchable_fields = ['mode', 'pickup_estimated', 'delivery_estimated']
+    for location_field in ['ship_from_location', 'ship_to_location', 'final_destination_location']:
+        for field in ['name', 'address_1', 'city', 'state', 'postal_code', 'country']:
+            searchable_fields.append(f'{location_field}__{field}')
+
+    search_fields = tuple(searchable_fields)
```

Comment:
Let's separate out `search_fields` and `filter_fields`. We don't need every field to be searchable, but we want most fields to be filterable. 
Think of search fields as fields that you would put in to a generic search box, like a reference number, a name, part of a location, etc. Filter fields are things you would have in a dropdown on a table. You should be able to have multiple filters applied. Without filter_fields there is no way for me to filter my results to only include locations with city=Greenville, state=SC
Take a look at django-filter: https://django-filter.readthedocs.io/en/latest/guide/usage.html?highlight=queryset#getting-started
You're already using this library - basically you need to declare a FilterSet that lays out all the filterable fields, then include that in the ViewSet.

Reply:
ok. Do you think I should have them in both? or just filter? As filter will exclude non-exact results which might be preferred

Reply:
We definitely want to have both search and filtering enabled. Search should only include a limited number of fields, whereas filtering should include as many filters as it makes sense to.